src/huggingface_hub/repocard.py

```diff
@@ -533,6 +534,7 @@ def metadata_save(local_path: Union[str, Path], data: Dict) -> None:
         readme.close()
 
 
+@_deprecate_positional_args(version="0.12")
 def metadata_eval_result(
     model_pretty_name: str,
```

Comment:
When using `_deprecate_positional_args` you can already place the `"*"` to make users aware it will be mandatory in next versions. The deprecate decorator handles positional argument to make same "keyword-based" in the meantime :) 

```suggestion
@_deprecate_positional_args(version="0.12")
def metadata_eval_result(
    *,
    model_pretty_name: str,
```